modulemd/modulemd-defaults-v1.c

```diff
@@ -1213,60 +1215,69 @@ modulemd_defaults_v1_merge (const gchar *module_name,
+
+  from_modified = modulemd_defaults_get_modified (MODULEMD_DEFAULTS (from));
+  into_modified = modulemd_defaults_get_modified (MODULEMD_DEFAULTS (into));
 
-  merged = modulemd_defaults_v1_new (module_name);
+  /* Start from a copy of "into" */
+  merged =
+    MODULEMD_DEFAULTS_V1 (modulemd_defaults_copy (MODULEMD_DEFAULTS (into)));
 
   /* Merge the default streams */
-  if (into->default_stream && !from->default_stream)
-    {
-      modulemd_defaults_v1_set_default_stream (
-        merged, into->default_stream, NULL);
-    }
-  else if (from->default_stream && !into->default_stream)
+  if (from->default_stream && !into->default_stream)
     {
       modulemd_defaults_v1_set_default_stream (
         merged, from->default_stream, NULL);
     }
-  else if (into->default_stream && from->default_stream)
+  else if (merged->default_stream && from->default_stream)
     {
-      if (g_str_equal (into->default_stream, DEFAULT_MERGE_CONFLICT))
+      if (g_str_equal (merged->default_stream, DEFAULT_MERGE_CONFLICT))
         {
           /* A previous pass over this same module encountered a merge
-           * conflict, so we need to propagate that.
+           * conflict, so keep it.
            */
-          modulemd_defaults_v1_set_default_stream (
-            merged, DEFAULT_MERGE_CONFLICT, NULL);
         }
-      else if (!g_str_equal (into->default_stream, from->default_stream))
+      else if (!g_str_equal (merged->default_stream, from->default_stream))
         {
-          /* They have conflicting default streams */
-          g_info ("Module stream mismatch in merge: %s != %s",
-                  into->default_stream,
-                  from->default_stream);
-          if (strict_default_streams)
+          if (from_modified > into_modified)
             {
-              g_set_error (error,
-                           MODULEMD_ERROR,
-                           MODULEMD_ERROR_VALIDATE,
-                           "Default stream mismatch in module %s: %s != %s",
-                           module_name,
-                           into->default_stream,
-                           from->default_stream);
-              return NULL;
+              modulemd_defaults_v1_set_default_stream (
+                merged, from->default_stream, NULL);
+            }
+          else if (from_modified == into_modified)
+            {
+              /* They have conflicting default streams */
+              g_info ("Module stream mismatch in merge: %s != %s",
+                      into->default_stream,
+                      from->default_stream);
+              if (strict_default_streams)
+                {
+                  g_set_error (
+                    error,
+                    MODULEMD_ERROR,
+                    MODULEMD_ERROR_VALIDATE,
+                    "Default stream mismatch in module %s: %s != %s",
+                    module_name,
+                    into->default_stream,
```

Comment:
```suggestion
                    merged->default_stream,
```

For consistency...

Reply:
This one is actually correct. It's a proof against future code refactoring resulting in `merged->default_stream` not matching `into->default_stream` anymore. We want to be sure to log what the inputs were that failed.

Reply:
Gotcha. I just find it bothersome that this is within a conditional block checking for equality against `merged->default_stream` while the error messages log `into->default_stream`.